boxes/generators/platonic.py

```diff
@@ -0,0 +1,174 @@
+
+        if not positive:
+            play = self.settings.play
+            f += play
+            s -= play
+            leftover -= play
+
+        self.edge(leftover, tabs=1)  # Whole point of this class
+
+        l1,l2 = self.fingerLength(self.settings.angle)
+        h = l1-l2
+
+        d = (bedBoltSettings or self.bedBoltSettings)[0]
+
+        for i in range(fingers):
+            if i != 0:
+                if not positive and bedBolts and bedBolts.drawBolt(i):
+                    self.hole(0.5 * s,
+                              0.5 * self.settings.thickness, 0.5 * d)
+
+                if positive and bedBolts and bedBolts.drawBolt(i):
+                    self.bedBoltHole(s, bedBoltSettings)
+                else:
+                    self.edge(s)
+
+            if positive and self.settings.style == "springs":
+                self.polyline(
+                    0, -90 * p, 0.8*h, (90 * p, 0.2*h),
+                    0.1 * h, 90, 0.9*h, -180, 0.9*h, 90,
+                    f - 0.6*h,
+                    90, 0.9*h, -180, 0.9*h, 90, 0.1*h,
+                (90 * p, 0.2 *h), 0.8*h, -90 * p)
+            else:
+                self.polyline(0, -90 * p, h, 90 * p, f, 90 * p, h, -90 * p)
+
+        self.edge(0, tabs=0)  # Whole point of this class
+
+class UnevenFingerJointEdgeCounterPart(UnevenFingerJointEdge):
+    """Uneven finger joint edge - other side"""
+    char = 'U'
+    description = "Uneven Finger Joint (opposing side)"
+    positive = False
+
+class Platonic(Boxes):
+    """Platonic solids generator"""
+
+    ui_group = "Unstable" # see ./__init__.py for names
+
+    SOLIDS = {
+        "Tetrahedron": (4, 3),
+        "Cube": (6, 4),
+        "Octahedron": (8, 3),
+        "Dodecahedron": (12, 5),
+        "Icosahedro": (20, 3),
+    }
+
+    def __init__(self):
+        Boxes.__init__(self)
+
+        # Uncomment the settings for the edge types you use
+        # use keyword args to set default values
+        self.addSettingsArgs(edges.FingerJointSettings)
+        # self.addSettingsArgs(edges.StackableSettings)
+        # self.addSettingsArgs(edges.HingeSettings)
+        # self.addSettingsArgs(edges.LidSettings)
+        # self.addSettingsArgs(edges.ClickSettings)
+        # self.addSettingsArgs(edges.FlexSettings)
+
+        # remove cli params you do not need
+        self.buildArgParser(x=90, sx="3*50", y=100, sy="3*50", h=100, hi=0)
+        # Add non default cli params if needed (see argparse std lib)
+        self.argparser.add_argument(
+            "--XX",  action="store", type=float, default=0.5,
+            help="DESCRIPTION")
+
+
+    def regularTrianagleWall(self, a, edges="uuuu", move=None):
+        self.regularPolygonWall(3, side=a, edges=edges, move=move)
+
+
+
+    def render(self):
+        # adjust to the variables you want in the local scope
+        x, y, h = self.x, self.y, self.h
+        a = x  # side length
+        t = self.thickness
+
+        solid = "Octahedron"
```

Comment:
See https://github.com/florianfesti/boxes/blob/master/boxes/generators/angledbox.py#L33 for an example on how to add a GUI parameter for such a selection.